h2/src/main/org/h2/command/Parser.java

```diff
@@ -7458,6 +7491,14 @@ private void parseTableColumnDefinition(CommandWithColumns command, Schema schem
                 if (readIf("AUTO_INCREMENT")) {
                     parseAutoIncrement(column);
                 }
+                if (database.getMode().useIdentityAsAutoIncrement) {
+                    if (readIf("NOT")) {
+                        read("NULL");
```

Comment:
1. `readIf(NOT)`, `read(NULL)`

2. I guess `column.setNullable(false);` should be invoked here.